### PR TITLE
Feat/chakra tag

### PR DESCRIPTION
**Pull request by Rykus0**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

closes #406 

## What is the new behavior?
Adds Chakra API for Tag

## Other information
Refactors Tag APIs to use a helper for building class names

### CODE REVIEW

**Comment by Rykus0** [comment]
[Chakra sandbox](https://codesandbox.io/s/basic-tag-chakra-q17lsd?file=/src/App.tsx)

File: packages/headless-styles/src/components/Tag/tagCSS.ts [hunk 1]
```diff
@@ -1,13 +1,12 @@
 import { createClassProp } from '../../utils/helpers'
-import { getDefaultTagOptions } from './shared'
+import { createTagSelectorClasses, getDefaultTagOptions } from './shared'
 import type { TagOptions } from './types'
 import styles from './tagCSS.module.css'
 
 export function getTagProps(options?: TagOptions) {
   const defaultOptions = getDefaultTagOptions(options)
   const { kind, size } = defaultOptions
```
Comment (caseybaggz) [inline packages/headless-styles/src/components/Tag/tagCSS.ts:8]:
```suggestion
```

File: packages/headless-styles/src/components/Tag/tagCSS.ts [hunk 2]
```diff
@@ -1,13 +1,12 @@
...
-  const kindClass = `${kind}Tag`
-  const sizeClass = `${size}Tag`
+  const { kindClass, sizeClass } = createTagSelectorClasses(kind, size)
```
Comment (caseybaggz) [inline packages/headless-styles/src/components/Tag/tagCSS.ts:9]:
```suggestion
  const { kindClass, sizeClass } = createTagSelectorClasses(deafultOptions.kind, defaultOptions.size)
```

File: packages/headless-styles/src/components/Tag/tagJS.ts [hunk 3]
```diff
@@ -1,17 +1,17 @@
...
 import type { TagOptions } from './types'
 import styles from './generated/tagCSS.module'
 
+type StylesKey = keyof typeof styles
 
 export function getJSTagProps(options?: TagOptions) {
   const { kind, size } = getDefaultTagOptions(options)
-  const sizeKey = `${size}Tag` as keyof typeof styles
-  const kindKey = `${kind}Tag` as keyof typeof styles
+  const { kindClass, sizeClass } = createTagSelectorClasses(kind, size)
```
Comment (caseybaggz) [inline packages/headless-styles/src/components/Tag/tagJS.ts:10]: Do the same here. No need to use const memory for 2 variables you are only using once. 😉

**Review by caseybaggz — APPROVED** [review]

**Pull request merged**